ocsp/responder.go

```diff
@@ -49,6 +51,49 @@ func (src InMemorySource) Response(request *ocsp.Request) (response []byte, pres
 	return
 }
 
+// SqliteSource represnts a source of OCSP responses backed by certdb
+type SqliteSource struct {
+	Accessor certdb.Accessor
+}
+
+// NewSqliteSource creates a new SqliteSource type and associated dbAccessor
+func NewSqliteSource(dbAccessor certdb.Accessor) Source {
+	return SqliteSource{
+		Accessor: dbAccessor,
+	}
+}
+
+// Response implements cfssl.ocsp.responder.Source, returning the OCSP response
+// with the expiration date furthest in the future
+func (src SqliteSource) Response(req *ocsp.Request) ([]byte, bool) {
+	if req == nil {
+		return nil, false
+	}
+
+	// Extract the AKI string from req.IssuerKeyHash
+	aki := hex.EncodeToString(req.IssuerKeyHash)
+
+	sn := req.SerialNumber
+	if sn == nil {
+		return nil, false
+	}
+	strSN := sn.String()
+
+	records, err := src.Accessor.GetOCSP(strSN, aki)
+	if err == nil || len(records) == 0 {
```

Comment:
Also, consider to add some logs. I think it's better to differentiate the errors here. if err != nil, log the error. If no record is returned, just return (nil, false)